src/sync/app_service_client.hpp

```diff
@@ -0,0 +1,50 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2020 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or utilied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+
+#ifndef APP_SERVICE_CLIENT_HPP
+#define APP_SERVICE_CLIENT_HPP
+
+#include <string>
+#include <realm/util/optional.hpp>
+#include <sync/generic_network_transport.hpp>
+
+namespace realm {
+namespace app {
+
+/// A class providing the core functionality necessary to make authenticated function call requests for a particular
+/// Stitch service.
+class AppServiceClient {
```

Comment:
All types with a virtual member function need to also have a virtual destructor, as otherwise deleting objects via a pointer to the base class won't call the destructor for the subclass.

Reply:
done